### PR TITLE
Document how to configure EsLint

### PR DESCRIPTION
**Pull request by teoljungberg**
- Sort the Table of Contents for the different languages and linters.

### CODE REVIEW

File: app/helpers/configuration_helper.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@ def javascript_ignore_url
     "https://raw.githubusercontent.com/thoughtbot/hound/master/config/style_guides/.jshintignore"
   end
 
+  def eslint_config_url
+    "https://raw.githubusercontent.com/thoughtbot/hound-eslint/master/config/default.json"
```
Comment (teoljungberg) [inline app/helpers/configuration_helper.rb]: EsLint needs a default configuration to run, this file is being added in https://github.com/thoughtbot/hound-eslint/pull/2

File: app/helpers/configuration_helper.rb [hunk 2]
```diff
@@ -1,25 +1,35 @@
...
+    config_url("thoughtbot/hound", "config/style_guides/javascript.json")
   end
 
   def javascript_ignore_url
-    "https://raw.githubusercontent.com/thoughtbot/hound/master/config/style_guides/.jshintignore"
+    config_url("thoughtbot/hound", "config/style_guides/.jshintignore.yml")
+  end
+
+  def eslint_config_url
+    config_url("thoughtbot/hound-eslint", "config/default.json")
```
Comment (teoljungberg) [inline app/helpers/configuration_helper.rb]: EsLint needs a default configuration to run, this file is being added in https://github.com/thoughtbot/hound-eslint/pull/2

**Comment by thorncp** [comment]
Seems fine to me. Do we want to use it internally a bit before merging this?

File: app/views/pages/configuration.haml [hunk 3]
```diff
@@ -182,6 +184,49 @@
...
+        to enable ESLINT style checking.
+
+        %code.code-block
+          :preserve
+            eslint:
+              enabled: true
+
+      %p
+        Hound uses
+        = link_to "EsLint",
```
Comment (JuanitoFatas) [inline app/views/pages/configuration.haml]: Should it be ESLint?

screenshot 2015-10-27 14 23 53

Comment (teoljungberg) [inline app/views/pages/configuration.haml]: Good catch!

File: app/views/pages/configuration.haml [hunk 4]
```diff
@@ -182,6 +184,49 @@
...
+
+      %p
+        For more information on the available rules in your
+        %em.code config_file
+        , you can read about them on the
+        = link_to "ESLint Rules Documentation",
+          "http://eslint.org/docs/rules/",
+          target: :blank
+
+
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: Extra newline?

File: app/helpers/configuration_helper.rb [hunk 5]
```diff
@@ -1,25 +1,35 @@
...
 
   def haml_config_url
-    "https://raw.githubusercontent.com/thoughtbot/hound/master/config/style_guides/haml.yml"
+    config_url("thoughtbot/hound", "config/style_guides/haml.yml")
+  end
+
+  private
+
+  def config_url(slug, config_file)
+    "https://raw.githubusercontent.com/#{slug}/master/#{config_file}"
```
Comment (gylaz) [inline app/helpers/configuration_helper.rb]: Nice extraction!

File: app/views/pages/configuration.haml [hunk 6]
```diff
@@ -4,21 +4,23 @@
...
-        = link_to "Go config", "#go"
+        = link_to "Hound config", "#configuration"
+      %li
+        = link_to "Javascript config", "#javascript"
       %li
         = link_to "Python config", "#python"
+      %li
+        = link_to "Ruby config", "#ruby"
+      %li
+        = link_to "SCSS config", "#scss"
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: Thoughts on dropping `config` word on all of these?

File: app/views/pages/configuration.haml [hunk 7]
```diff
@@ -182,6 +184,48 @@
             javascript:
               enabled: false
 
+    %article#eslint
+      %h3 ESLint (beta)
+
+      %p
+        Add the following code to your
+        %em.code .hound.yml
+        to enable ESLINT style checking.
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: `ESLINT` to `ESLint`?

File: app/views/pages/configuration.haml [hunk 8]
```diff
@@ -182,6 +184,48 @@
...
+        %code.code-block
+          :preserve
+            eslint:
+              enabled: true
+
+      %p
+        Hound uses
+        = link_to "ESLint",
+          "http://eslint.org/",
+          target: :blank
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: Fit on one line?

**Comment by gylaz** [comment]
Looks good once https://github.com/thoughtbot/hound-eslint/pull/2 lands.

**Pull request merged**